src/scanner.c

```diff
@@ -1810,6 +1810,81 @@ void GetNumber ( UInt StartingStatus )
 }
 
 
+/*******************************************************************************
+ **
+ *F  GetEscapedChar()   . . . . . . . . . . . . . . . . get an escaped character
+ **
+ **  'GetEscapedChar' reads an escape sequence from the current input file into
+ **  the variable *dst.
+ **
+ */
+static inline Char GetOctalDigits( void )
+{
+    Char c;
+
+    if ( *TLS(In) < '0' || *TLS(In) > '7' )
+        SyntaxError("Expecting octal digit");
+    c = 8 * (*TLS(In) - '0');
+    GET_CHAR();
+    if ( *TLS(In) < '0' || *TLS(In) > '7' )
+        SyntaxError("Expecting octal digit");
+    c = c + (*TLS(In) - '0');
+
+    return c;
+}
+
+Char GetEscapedChar( void )
+{
+  Char c;
+
+  c = 0;
+
+  if ( *TLS(In) == 'n'  )       c = '\n';
+  else if ( *TLS(In) == 't'  )  c = '\t';
+  else if ( *TLS(In) == 'r'  )  c = '\r';
+  else if ( *TLS(In) == 'b'  )  c = '\b';
+  else if ( *TLS(In) == '>'  )  c = '\01';
+  else if ( *TLS(In) == '<'  )  c = '\02';
+  else if ( *TLS(In) == 'c'  )  c = '\03';
+  else if ( *TLS(In) == '"'  )  c = '"';
+  else if ( *TLS(In) == '\\' )  c = '\\';
+  else if ( *TLS(In) == '\'' )  c = '\'';
+  else if ( *TLS(In) == '0'  ) {
+    /* from here we can either read a hex-escape or three digit
+       octal numbers */
+    GET_CHAR();
+    if (*TLS(In) == 'x') {
+        GET_CHAR();
+        if (!IsHexDigit(*TLS(In))) {
+            SyntaxError("Expecting hexadecimal digit");
+        }
+        c = 16 * CharHexDigit(*TLS(In));
+        GET_CHAR();
+        if (!IsHexDigit(*TLS(In))) {
+            SyntaxError("Expecting hexadecimal digit");
+        }
+        c += CharHexDigit(*TLS(In));
+    } else if (*TLS(In) >= '0' && *TLS(In) <= '7' ) {
+        c += GetOctalDigits();
+    } else {
+        SyntaxError("Expecting hexadecimal escape, or two more octal digits");
+    }
+  } else if ( *TLS(In) >= '1' && *TLS(In) <= '7' ) {
+    /* escaped three digit octal numbers are allowed in input */
+    c = 64 * (*TLS(In) - '0');
+    GET_CHAR();
+    c += GetOctalDigits();
+  } else {
+      /* This warning is currently disabled for backwards compatibility */
```

Comment:
Perhaps elaborate a bit, and/or reference this PR. E.g.:
"This warning is currently disabled for backwards compatibility: It turns out there are some packages which still rely on this."